engine/src/main/scala/cromwell/engine/workflow/workflowstore/WorkflowStoreHeartbeatWriteActor.scala

```diff
@@ -40,21 +40,37 @@ case class WorkflowStoreHeartbeatWriteActor(workflowStoreAccess: WorkflowStoreAc
     *
     * @return the number of elements processed
     */
-  override protected def process(data: NonEmptyVector[(WorkflowId, OffsetDateTime)]): Future[Int] = instrumentedProcess {
-    val heartbeatDateTime = OffsetDateTime.now()
-    val processFuture = workflowStoreAccess.writeWorkflowHeartbeats(data, heartbeatDateTime)
-    processFuture transform {
-      // Track the `Try`, and then return the original `Try`. Similar to `andThen` but doesn't swallow exceptions.
-      _ <| trackRepeatedFailures(heartbeatDateTime, data.length)
+  override protected def process(data: NonEmptyVector[WorkflowStoreWriteHeartbeatCommand]): Future[Int] = instrumentedProcess {
+    val now = OffsetDateTime.now()
+    val workflowsIdsHavingStaleHeartbeats: Seq[WorkflowId] = data.collect {
+      case h if JDuration.between(h.heartbeatTime, now).toNanos >= failureShutdownDuration.toNanos => h.workflowId
+    }
+
+    if (workflowsIdsHavingStaleHeartbeats.isEmpty) {
+      val processFuture = workflowStoreAccess.writeWorkflowHeartbeats(data.map { h => (h.workflowId, h.submissionTime) }, now)
+      processFuture transform {
+        // Track the `Try`, and then return the original `Try`. Similar to `andThen` but doesn't swallow exceptions.
+        _ <| trackRepeatedFailures(now)
+      }
+    } else {
+      log.error(String.format(
+        "Shutting down %s as %s stale workflow heartbeats (older than %s) were found. Workflow ids with stale heartbeats: %s",
+        workflowHeartbeatConfig.cromwellId,
+        workflowsIdsHavingStaleHeartbeats.size.toString,
+        failureShutdownDuration.toString(),
+        workflowsIdsHavingStaleHeartbeats mkString ", "
+      ))
+      terminator.beginCromwellShutdown(WorkflowStoreHeartbeatWriteActor.Shutdown)
```

Comment:
![image](https://user-images.githubusercontent.com/13006282/156785381-9eaf6e54-5716-45ec-a5e0-27f0e5237868.png)

Reply:
It'll be back